### DISCUSSION TITLE
Open tab programmatically without using state

### DISCUSSION THREAD

**Discussion by godoom** [radix-ui/primitives#2154] [Help]
How would i use defaultValue, value, or onValueChange props of the radix Tabs component so that I change the tab programmatically when some could in my react context runs? I hav a function in context that sets the tab on a page to switch to a different tab. I dont want to use state because the tabs are in a server component.

**Comment by benoitgrelard** [comment]
If you want to update which tab is showing, you have no choice but to use state, that is how interactivity works in React. Not everything needs to be a server component either.
